src/java.base/share/classes/jdk/internal/foreign/SegmentFactories.java

```diff
@@ -203,8 +203,9 @@ private static long allocateNativeInternal(long byteSize, long byteAlignment, Me
             throw new OutOfMemoryError();
         }
         // Always allocate at least some memory so that zero-length segments have distinct
-        // non-zero addresses.
-        alignedSize = Math.max(Long.BYTES, alignedSize);
+        // non-zero addresses. If we are initializing the allocated memory, then use a minimum
+        // size of 8 because we init the memory with longs.
+        alignedSize = Math.max((init ? Long.BYTES : 1), alignedSize);
```

Comment:
Is there a test case where `byteSize` is zero? In that case `alignedSize` would also be zero and the old code would update it to 8. How does reducing `alignedSize` to 1 help (when `init` is false)?

Reply:
We should allow that change to flow downstream naturally. I see no urgency to make that change now.

Reply:
I disagree: I don't think it's better and I don't expect a conflict when the upstream change arrives here. I also disagree with the upstream suggestion.